test/unit/repositories/test_catalogue_category.py

```diff
@@ -353,8 +358,14 @@ def test_create_with_duplicate_name_within_parent(test_helpers, database_mock, c
             "_id": CustomObjectId(catalogue_category.parent_id),
         },
     )
-    # Mock `count_documents` to return 1 (duplicate catalogue category found within the parent catalogue category)
-    test_helpers.mock_count_documents(database_mock.catalogue_categories, 1)
+    # Mock `find_one` to return duplicate catalogue category found
+    test_helpers.mock_find_one(
+        database_mock.catalogue_categories,
+        {
+            **CATALOGUE_CATEGORY_INFO,
+            "_id": CustomObjectId(catalogue_category.parent_id),
```

Comment:
Shouldn't the ID of the duplicate catalogue category be different to that of the parent (line 358)? It's testing a duplicate name within the parent so it should have a different ID but its parent ID should be the ID of the parent (line 358)?